src/vm/mod.rs

```diff
@@ -255,7 +256,7 @@ fn eval_all(
             let try_define = global_context.execute(|context| {
                 let mut call_stack = CallStack::new();
                 let mut env = Environment::new(
-                    context, contract_context, &mut call_stack, Some(publisher.clone()), Some(publisher.clone()));
+                    context, contract_context, &mut call_stack, Some(publisher.clone()), Some(publisher.clone()), None);
```

Comment:
These `None` values should take the passed `sponsor`, that way the contract initialization code can use `tx-sponsor?` as well.